translator/opencensus/oc_to_internal.go

```diff
@@ -65,26 +63,35 @@ func ocToInternal(td consumerdata.TraceData) data.TraceData {
 	// Count the number of spans that have nil Resource and need to be combined
 	// in one slice.
 	combinedSpanCount := 0
+	combinedResources := 0
 	for _, ocSpan := range td.Spans {
 		if ocSpan == nil {
 			// Skip nil spans.
 			continue
 		}
 		if ocSpan.Resource == nil {
 			combinedSpanCount++
+		} else {
+			combinedResources++
 		}
 	}
+	// Total number of resources is equal with:
+	// 1 (for all spans with nil resource) + numSpansWithResource (combinedResources).
+	traceData.SetResourceSpans(data.NewResourceSpansSlice(combinedResources + 1))
+	ocNodeResourceToInternal(td.Node, td.Resource, traceData.ResourceSpans().Get(0))
 
-	// Allocate a slice for spans that need to be combined into one ResourceSpans.
-	combinedSpans := data.NewSpanSlice(combinedSpanCount)
-	resourceSpans.SetInstrumentationLibrarySpans([]*data.InstrumentationLibrarySpans{
-		data.NewInstrumentationLibrarySpans(data.NewInstrumentationLibrary(), combinedSpans)})
+	// Allocate a slice for spans that need to be combined into first ResourceSpans.
+	traceData.ResourceSpans().Get(0).SetInstrumentationLibrarySpans(data.NewInstrumentationLibrarySpansSlice(1))
```

Comment:
Can we assign `traceData.ResourceSpans().Get(0)` to temp var to avoid repetition?

Reply:
Done.